pollux/lib/core/src/main/scala/io/iohk/atala/pollux/core/service/CredentialServiceImpl.scala

```diff
@@ -1048,22 +1048,25 @@ private class CredentialServiceImpl(
         .mapError(RepositoryError.apply)
       size = currentStatusList.size
       lastUsedIndex = currentStatusList.lastUsedIndex
-      // TODO: concurrency issue
-      statusListToBeUsed <-
-        if lastUsedIndex < size then ZIO.succeed(currentStatusList)
-        else
-          credentialStatusListRepository
-            .createNewForTheWallet(jwtIssuer, statusListRegistryUrl)
-            .mapError(RepositoryError.apply)
-
-      _ <- credentialStatusListRepository
-        .allocateSpaceForCredential(
-          issueCredentialRecordId = record.id,
-          credentialStatusListId = statusListToBeUsed.id,
-          statusListIndex = statusListToBeUsed.lastUsedIndex + 1
-        )
-        .mapError(RepositoryError.apply)
+      sem <- Semaphore.make(permits = 1)
```

Comment:
Does this create new semaphore on every call? Should semaphore instance be shared so permit is enforced for all calls? :thinking:

Reply:
I think the semaphore instance should be shared. Where to create it might depend on how the permit should be shared. If you need a global permit then I have done something similar [here](https://github.com/hyperledger-labs/open-enterprise-agent/blob/8f6af25a8eafd27d5017096da64f89188354a2ca/prism-agent/service/wallet-api/src/main/scala/io/iohk/atala/agent/walletapi/service/ManagedDIDServiceImpl.scala#L33).